### ISSUE TITLE
`single_range_in_vec_init` produces invalid suggestions when using explict `Range { start:x, end:y }` vector element

### ISSUE THREAD

**Issue by paulmenage** [rust-lang/rust-clippy#16042]
### Summary

I saw this output from clippy:
```
error: a `Vec` of `Range` that is only one element
   --> odd/src/cmd/dap_server/server/core_data.rs:615:17
    |
615 |     let input = vec![Range { start: 0, end: 5 }];
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_range_in_vec_init
    = note: `-D clippy::single-range-in-vec-init` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::single_range_in_vec_init)]`
help: if you wanted a `Vec` that contains the entire range, try
    |
615 -     let input = vec![Range { start: 0, end: 5 }];
615 +     let input = (start: 0..end: 5).collect::<std::vec::Vec<u64>>();
    |
help: if you wanted a `Vec` of len end: 5, try
    |
615 -     let input = vec![Range { start: 0, end: 5 }];
615 +     let input = vec![start: 0; end: 5];
    |
```
Probably if the vector element is explicitly using the `Range{}` constructor it should do one of:
- ignore it, it's probably what was actually intended
- strip out the `start:` and `end:` in the suggestions
- leave the `Range{}` constructor intact (in the first suggestion, at least), i.e. `let input = Range { start: 0, end: 5}.collect::<...>()`

### Reproducer

I tried this code:

```rust
let input = vec![Range { start: 0, end: 5 }];
```

I expected to see this happen:
```
help: if you wanted a `Vec` that contains the entire range, try
    |
615 -     let input = vec![Range { start: 0, end: 5 }];
615 +     let input = (0..5).collect::<std::vec::Vec<u64>>();
```
(or just ignore it as most likely a false-positive)

Instead, this happened:

```
help: if you wanted a `Vec` that contains the entire range, try
    |
615 -     let input = vec![Range { start: 0, end: 5 }];
615 +     let input = (start: 0..end: 5).collect::<std::vec::Vec<u64>>();
```

### Version

```text
rustc 1.93.0-nightly (f15a7f385 2025-11-04)
binary: rustc
commit-hash: f15a7f38580ddbdc1a23909dd05cf6cc6d9f3919
commit-date: 2025-11-04
host: aarch64-apple-darwin
release: 1.93.0-nightly
LLVM version: 21.1.3
```
